x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/inference/action/InferenceAction.java

```diff
@@ -152,16 +159,20 @@ public void writeTo(StreamOutput out) throws IOException {
             // in version ML_INFERENCE_REQUEST_INPUT_TYPE_ADDED the input type enum was added, so we only want to write the enum if we're
             // at that version or later
             if (out.getTransportVersion().onOrAfter(TransportVersions.ML_INFERENCE_REQUEST_INPUT_TYPE_ADDED)) {
-                out.writeEnum(getInputTypeToWrite(out.getTransportVersion()));
+                out.writeEnum(getInputTypeToWrite(inputType, out.getTransportVersion()));
             }
         }
 
-        private InputType getInputTypeToWrite(TransportVersion version) {
-            // in version ML_INFERENCE_REQUEST_INPUT_TYPE_UNSPECIFIED_ADDED the UNSPECIFIED value was added, so if we're before that
-            // version other nodes won't know about it, so set it to INGEST instead
-            if (version.before(TransportVersions.ML_INFERENCE_REQUEST_INPUT_TYPE_UNSPECIFIED_ADDED) && inputType == InputType.UNSPECIFIED) {
+        // default for easier testing
+        static InputType getInputTypeToWrite(InputType inputType, TransportVersion version) {
+            if (version.before(TransportVersions.ML_INFERENCE_REQUEST_INPUT_TYPE_UNSPECIFIED_ADDED)
+                && validEnumsBeforeUnspecifiedAdded.contains(inputType) == false) {
```

Comment:
If we're before the version when the unspecified enum was added and `inputType` is not one of `ingest` or `search`, then we'll revert to `ingest` because the node does not know about the other values.